Podstawy programowania/Ćwiczenia powtórzeniowe/39-R_Cwiczenie_powtorzeniowe1.py

```diff
@@ -0,0 +1,41 @@
+# Zadanie 1
+# Rozwiązać równanie ax+b=0.
+
+def get_number(name):
+    number = input("Podaj liczbę " + name + ": ")
+    while True:
+      try:
+          number = float(number)
+          break
+      except:
+          number = input("Błąd, podaj liczbę: ")
+    return number
```

Comment:
Wydaje mi się, że możnaby to napisać krócej:
```
    while True:
      try:
          return number = float(number)
      except:
          number = input("Błąd, podaj liczbę: ")

```
Chodzi o to, żeby zwrócić "dobrą" liczbę. Próba konwersji do float i zajdzie najpierw, zanim nastąpi zwrot, więc jeżeli będzie miał być błąd, to będzie. A jeżeli będzie, to nastąpi przejście do bloku except. 

Ale gdybyś zmieniał, to porządnie to przetestuj.

Reply:
podkreśla mi, że "float(number)" unreachable więc chyba zostawię jak jest, bo działa...

Reply:
while True:
      try:
          return float(number)
      except:
          number = input("Błąd, podaj liczbę: ")

Reply:
Przedyskutowaliśmy to online :)